app/Http/Controllers/Api/SitesController.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Site;
+use App\Plot;
+use App\Plant;
+use App\Measurement;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+
+class SitesController extends Controller
+{
+    protected static $quadrants = [
+        'Southwest',
+        'Northwest',
+        'Southeast',
+        'Northeast',
+    ];
+
+    /**
+     * Returns a user's sites and shared sites, as well as the associated plots, plants, and measurements.
+     * @param Request $request
+     * @return JsonResponse|Response
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function download(Request $request)
+    {
+        // Authorization taken care of by Sanctum
+
+        $user = $request->user();
+
+        $sites = Site::with([
+            'species',
+            'shrubs',
+            'plots' => function ($query) {
+                $query->with([
+                    'plants' => function ($query) {
+                        $query->with(['measurements', 'species']);
+                    },
+                ]);
+            }
+        ])
+            ->withShared($user)
+            ->orWhere('user_id', $user->id)
+            ->get();
+
+        return $this->success($sites);
+    }
+
+    /**
+     * Takes in one or more sites from the app and updates the server with it
+     * @param Request $request
+     * @return string
+     * @throws \Throwable
+     */
+    public function uploadSites(Request $request)
+    {
+        DB::beginTransaction();
+        $user = $request->user();
+        if ($request->site) {
+            $sites = [$request->site];
+        } else {
+            $sites = $request->sites;
+        }
+
+        $sitesApi = new SitesApi();
+        foreach ($sites as $site) {
+            $validation = $this->validateSite($site);
+            if ($validation != 'validated successfully') {
+                return $validation;
+            }
```

Comment:
The duplication issue still remains since we don't rollback transactions. But since you've extracted validation to its own function, this can be fixed easily:

```suggestion
        $sitesApi = new SitesApi();
        
        foreach($sites as $site) {
            $validation = $this->validateSite($site);
            if ($validation != 'validated successfully') {
                return $validation;
            }
        }
        
        DB::beginTransaction();
        foreach ($sites as $site) {
```

By validating all sites first, we can return any errors before we even start adding anything to the database.